### PR TITLE
feat(structured data): remaining front API endpoints

### PR DESCRIPTION
**Pull request by fontanierh**
this is a lot of pedestrian code to implement the public front API endpoints for databases.

### CODE REVIEW

File: front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "The database id is missing.",
+      },
+    });
+  }
+
+  switch (req.method) {
+    case "GET":
```
Comment (spolu) [inline front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts:68]: we don't check for builder status in other v1 endpoints?

**Review by spolu — APPROVED** [review]

File: front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts (hunk 1 above)
Comment (fontanierh) [reply front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts:68]: Api key means builder no?

Comment (spolu) [reply front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts:68]: For now but maybe not tomorrow

Comment (spolu) [reply front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts:68]: #defense_in_depth

Comment (fontanierh) [reply front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts:68]: We don't do it anywhere else, but if you want I can change all the v1 endpoints to add the check ?

Comment (fontanierh) [reply front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts:68]: Most likely preferrable as a separate PR though, wdyt ?

Comment (spolu) [reply front/pages/api/v1/w/[wId]/data_sources/[name]/databases/[id]/index.ts:68]: If we don’t do it anywhere else then fine to keep as is 👍 as we’ll know we’ll have to do it when the time comes

**Pull request merged**